src/Text/Pandoc/Readers/Markdown.hs

```diff
@@ -1687,7 +1687,8 @@ superscript = do
     char '^'
     mconcat <$> many1Till (do notFollowedBy spaceChar
                               notFollowedBy newline
-                              inline) (char '^'))
+                              inline) ((char '^') <|> (do guardEnabled Ext_mmd_short_superscript
```

Comment:
Lines are best kept shorter than 80 chars, as it often helps with readability. Maybe we could define a local parser `end` which succeeds if the end of the superscript is reached?